MinecraftClient/McTcpClient.cs

```diff
@@ -1278,7 +1284,35 @@ public void OnSpawnPlayer(int entityID, Guid uuid, Location location, byte Yaw,
             Entity playerEntity = new Entity(entityID, EntityType.Player, location, uuid, playerName);
             OnSpawnEntity(playerEntity);
         }
-
+        
+        public void OnEntityEquipment(int entityid, int slot, Item item)
+        {
+            foreach (ChatBot bot in bots.ToArray())
+            {
+                try
+                {
+                    if (entities.ContainsKey(entityid))
+                        bot.OnEntityEquipment(entities[entityid], slot, item);
+                }
+                catch (Exception e)
+                {
+                    try
+                    {
+                        if (entities.ContainsKey(entityid))
+                            bot.OnEntityEquipment(entities[entityid], slot, new Item(0, 0, null));

```

Comment:
When I remove the item from my hand, the item.Type does not issue Air

Reply:
Ok, in that case I think fixing that in order to have type back to Air when item is removed is better that sending the event twice because this might confuse bots and produce unreliable results.

Reply:
Could you fix it I haven't found another solution

Reply:
In your bot, check if item is invalid and treat it as empty instead of crashing/throwing an exception.